addons/sale/static/src/js/reconciliation/reconciliation_action.js

```diff
@@ -0,0 +1,262 @@
+odoo.define('sale.ReconciliationClientAction', function (require) {
+"use strict";
+
+var ReconciliationClientAction = require('account.ReconciliationClientAction');
+var core = require('web.core');
+var fieldUtils = require('web.field_utils');
+var Dialog = require('web.view_dialogs');
+
+var _t = core._t;
+var saleOrderSelectViewId;
+var saleOrderSearchViewId;
+
+
+ReconciliationClientAction.StatementAction.include({
+    custom_events: _.extend({}, ReconciliationClientAction.StatementAction.prototype.custom_events, {
+        reconcile_with_sale_order: '_onReconcileWithSaleOrder',
+    }),
+
+    /**
+     * @override
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        if (!saleOrderSelectViewId) {
```

Comment:
i think that this should go in willStart, with a $.when.  This means that you will have a guarantee that these calls are completed before you use them